crates/autopilot/src/run_loop.rs

```diff
@@ -0,0 +1,77 @@
+use anyhow::{anyhow, Context, Result};
+use chrono::Utc;
+use model::auction::{Auction, AuctionId};
+use shared::current_block::CurrentBlockStream;
+use std::time::Duration;
+
+use crate::{
+    database::Postgres, driver_api::Driver, driver_model::solve,
+    solvable_orders::SolvableOrdersCache,
+};
+
+const SOLVE_TIME_LIMIT: Duration = Duration::from_secs(15);
+
+pub struct RunLoop {
+    solvable_orders_cache: SolvableOrdersCache,
+    database: Postgres,
+    drivers: Vec<Driver>,
+    current_block: CurrentBlockStream,
+}
+
+impl RunLoop {
+    pub async fn run_forever(&self) -> ! {
+        loop {
+            match self.single_run().await {
+                Ok(()) => tracing::debug!("single run ok"),
+                Err(err) => tracing::error!(?err, "single run err"),
+            }
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        }
+    }
+
+    async fn single_run(&self) -> Result<()> {
+        let auction = match self.solvable_orders_cache.current_auction() {
+            Some(auction) => auction,
+            None => {
+                tracing::debug!("no current auction");
+                return Ok(());
+            }
+        };
+        let id = self
+            .database
+            .replace_current_auction(&auction)
+            .await
+            .context("replace_current_auction")?;
+
+        let _responses = self.solve(&auction, id).await;
+
+        // TODO:
+        // - find winner
+        // - send execute to winner
+        // - store per auction info
```

Comment:
cc @ennmichael as this relates to what we discussed in Slack about:

>  would even say that for 4 - since we plan on running it remotely, I would even not call `/execute` on the driver at all (in case it is misconfigured).
> 
> The tricky part is that then we don’t have calldata - so maybe there should be a new route for building calldata? I’m a little fuzzy on this though. Note that we can punt this specific decision until we start running `driver` -s out of our infra.

The `/reveal` route proposed by @vkgnosis can be used for the shadow `autopilot` as well (for getting calldata from drivers without asking them to execute the transactions onchain).

Reply:
We decided to not add a `/reveal` route so resolving this.

Instead, we will append unique IDs to calldata as a way to disambiguate transactions from order executions per auction.